src/main/java/net/glowstone/generator/structures/GlowDungeon.java

```diff
@@ -141,19 +141,19 @@ public boolean generate(World world, Random random, StructureBoundingBox genBoun
                 if (builder.getBlockState(new Vector(x, 1, z)).getType() == Material.AIR) {
                     BlockFace face = null;
                     int solidBlocksCount = 0;
-                    if (builder.getBlockState(new Vector(x - 1, 1, z)).getType().isSolid()) {
+                    if (builder.getBlockState(new Vector(x - 1, 1, z)).getType().equals(Material.COBBLESTONE)) {
```

Comment:
I think using `==` on the Material enum (`getType()` and `Material.COBBLESTONE`) would be better than using the equals() method, which simply just returns `this==other`.